### PR TITLE
Artk-135

### PR DESCRIPTION
**Pull request by ThorstenBux**
Fix in the make file for clang and some improvement in the Configure script so enhance UX.
Please be aware that it goes together with a change in the artoolkit-doc with the same branch name.

### CODE REVIEW

File: Configure [hunk 1]
```diff
@@ -112,15 +113,16 @@ MACHINE=`uname -m`
 if [ "$E" = "Linux" ]
 then
     SYSTEM="linux-$MACHINE"
-    echo "Do you want to use the Clang compiler? (y or n)"
+    echo "\nThis script builds ARToolKit libraries, utilities and examples. For details on dependencies and further information please visit http://artoolkit.org/documentation/doku.php?id=8_Advanced_Topics:build_artoolkit \n\n"
```
Comment (philip-lamb) [inline Configure]: Change "builds" to "configures". 'Configure' doesn't build anything.

Comment (ThorstenBux) [inline Configure]: done

File: Configure [hunk 2]
```diff
@@ -112,15 +113,16 @@ MACHINE=`uname -m`
...
+    echo "Do you want to use the GNU (y) (or clang (n)) compiler?"
```
Comment (philip-lamb) [inline Configure]: Confusing, and as I already said, no need to change the sense of the question... keep 'y' as the option for Clang. Perhaps: "Compiler is gcc and g++ with libstdc++. Do you want to use the Clang compiler with libc++? (y or n)".

Comment (ThorstenBux) [inline Configure]: done

File: examples/stereo/Makefile.in [hunk 3]
```diff
@@ -53,7 +53,7 @@ AR_CPPFLAGS = -I$(AR_HOME)/include
 AR_LDFLAGS = -L$(AR_HOME)/lib/@SYSTEM@ -L$(AR_HOME)/lib
 
 OSG_LIBPATH = /usr/lib
-OSG_LIBS = -losg -lOpenThreads -losgDB -losgFX -losgGA -losgViewer -losgUtil -lstdc++
+OSG_LIBS = -losg -lOpenThreads -losgDB -losgFX -losgGA -losgViewer -losgUtil @STDCXXLIB@
 ifeq "$(UNAME)" "Darwin"
```
Comment (philip-lamb) [inline examples/stereo/Makefile.in]: There are several more of these which need to be changed:

Zen:~/Documents/Development/Projects/ARToolKit5/examples$ find . -name "Makefile.in" | xargs grep -rE --  "-lstdc\+\+"
./nftBook/Makefile.in:OSG_LIBS = -losg -lOpenThreads -losgDB -losgFX -losgGA -losgViewer -losgUtil -lstdc++
./nftBook/Makefile.in:    OSG_LIBS = -framework osg -framework OpenThreads -framework osgDB -framework osgFX -framework osgGA -framework osgViewer -framework osgUtil -lstdc++
./optical/Makefile.in:OSG_LIBS = -losg -lOpenThreads -losgDB -losgFX -losgGA -losgViewer -losgUtil -lstdc++
./optical/Makefile.in:    OSG_LIBS = -framework osg -framework OpenThreads -framework osgDB -framework osgFX -framework osgGA -framework osgViewer -framework osgUtil -lstdc++
./opticalStereo/Makefile.in:OSG_LIBS = -losg -lOpenThreads -losgDB -losgFX -losgGA -losgViewer -losgUtil -lstdc++
./opticalStereo/Makefile.in:    OSG_LIBS = -framework osg -framework OpenThreads -framework osgDB -framework osgFX -framework osgGA -framework osgViewer -framework osgUtil -lstdc++
./simpleOSG/Makefile.in:OSG_LIBS = -losg -lOpenThreads -losgDB -losgFX -losgGA -losgViewer -losgUtil -lstdc++
./simpleOSG/Makefile.in:    OSG_LIBS = -framework osg -framework OpenThreads -framework osgDB -framework osgFX -framework osgGA -framework osgViewer -framework osgUtil -lstdc++
./simpleVRML/Makefile.in:    -lopenvrml -lopenvrml-gl -lstdc++ -ljpeg -lpng -lz -lm \
./stereo/Makefile.in:OSG_LIBS = -losg -lOpenThreads -losgDB -losgFX -losgGA -losgViewer -losgUtil -lstdc++
./stereo/Makefile.in:    OSG_LIBS = -framework osg -framework OpenThreads -framework osgDB -framework osgFX -framework osgGA -framework osgViewer -framework osgUtil -lstdc++

Comment (ThorstenBux) [inline examples/stereo/Makefile.in]: Well, they didn't fail while building that is why I did not look into them

Comment (ThorstenBux) [inline examples/stereo/Makefile.in]: Done

Comment (ThorstenBux) [inline examples/stereo/Makefile.in]: Forgot to save one. That is in now.

Comment (ThorstenBux) [inline examples/stereo/Makefile.in]: Don't merge jet
There is the nftSimple that fails too

**Comment by ThorstenBux** [comment]
Alright all compile options are now included and OSG examples are excluded from build if building with clang.

Ready to merge!

File: examples/stereo/Makefile.in (hunk 3 above)
Comment (BoiseBaked) [inline examples/stereo/Makefile.in]: Only one thing we may want to remember for a future sprint task is to move all Linux related build configuration, helper files and build scripts out of the repo root directory to under a folder name "Linux" or "LinuxBuild" or something.

File: Configure [hunk 4]
```diff
@@ -1,4 +1,5 @@
 #!/bin/sh
+# tputcolors
```
Comment (philip-lamb) [inline Configure]: What is the intent and function of this?

Comment (ThorstenBux) [inline Configure]: left over from when I coloured the information about clang. But as we agreed to remove this that should go as well.

Thanks for reminding me.

I removed it.

**Pull request merged**